testsuite/features/step_definitions/command_steps.rb

```diff
@@ -208,7 +208,14 @@
 
 When(/^I query latest Salt changes on ubuntu system "(.*?)"$/) do |host|
   node = get_target(host)
-  salt = $use_salt_bundle ? "venv-salt-minion" : "salt"
+  salt =
+    if ENV['PROVIDER'] == PROVIDER
```

Comment:
This line is very confuse, not at all clear for maintainability.
What is `PROVIDER`? Also we should try to avoid calls to system variables inside steps, instead just load it once in env.rb, as `$provider = ENV['PROVIDER']` then you use `$provider` inside the steps.